No marginal space validation added

ajithkumar-maragathavel opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
No marginal space validation added for **identifier** field in cms block grid in-line edit.
Magento admin path to CMS Block admin grid :
Content -> Elements -> Blocks

This validation allows us to prevent users from entering leading and trailing spaces in Identifier field using in-line edit in the Grid.

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Enter text with leading and trailing spaces in Identifier field, and it will display error message.
2. Enter text without leading and trailing spaces in Identifier field and it won't display error message.


### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

Commit:
MC-29167: CatalogWidget products list doesn't work with anchor category

Commit:
MC-29167: CatalogWidget products list doesn't work with anchor category

Commit:
MC-29167: CatalogWidget products list doesn't work with anchor category

Commit:
MC-29745: Bundle child product with qty increments error

Commit:
Merge remote-tracking branch 'origin/2.4-develop' into MC-29745

SeruyV committed:
MC-21347: Remove Customer module's dependency on Review

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2ce into MC-29167

SeruyV committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-21347

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Customer/Controller/Adminhtml/IndexTest.php

SeruyV committed:
MC-21347: Remove Customer module's dependency on Review

SeruyV committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-21347

zakdma committed:
MC-29864: Incorrect MC-13493 behaviour when ElasticSearch enabled

ajithkumar-maragathavel committed:
No marginal space validation added

rogyar commented:
Hi @ajithkumar-maragathavel. Thank you for your collaboration. 

According to the [Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html), changes like this one should be covered with auto-tests. I would recommend covering the current case with MFTF test. 

Thank you!

SeruyV committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-21347

SeruyV committed:
MC-21347: Remove Customer module's dependency on Review

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2ce into MC-29167

Commit:
MC-30260: CatalogWidget products list doesn't display products from the children categories of 2nd level and up

Commit:
MC-30260: CatalogWidget products list doesn't display products from the children categories of 2nd level and up

zakdma committed:
MC-25138: 24/36 product limit on Category page occur fatal

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-21347

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2ce into MC-30260

Commit:
Issue-25968 - Added additional checking for returning needed variable type

lbajsarowicz committed:
REFACTOR: Extract Action Groups to separate files (MFTF best practices)

DianaRusin committed:
MC-25250: DropDown attribute with label 'select' break product edit page

DianaRusin committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25250

DianaRusin committed:
MC-25250: DropDown attribute with label 'select' break product edit page

lbajsarowicz committed:
REFACTOR: Extract Action Groups to separate files (MFTF best practices)

magento-engcom-team committed:
ENGCOM-6565: MFTF: Extract Action Groups to separate files for dev/tests #26322

 - Merge Pull Request magento/magento2#26322 from lbajsarowicz/magento2:mftf/action-groups
 - Merged commits:
   1. 710d6920bbf7ef6569c74d91989f19768ffd6483
   2. 0e4f6ce0d35f90e8a2cd371d63d98f15ce85605e

tejash-wagento committed:
Add to Compare link not showing in mobile view under 640px

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25138

svitja committed:
MC-25254: [ElasticSearch] New product attribute requires full search reindex to be searchable on Storefront

svitja committed:
MC-25254: [ElasticSearch] New product attribute requires full search reindex to be searchable on Storefront

svitja committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25254

Commit:
Issue-25968 - Added adjustments after review

Commit:
Issue-25968 - Small adjustments after code review

svitja committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25254

atwixfirster committed:
magento/magento2#: Unit test for \Magento\AdminNotification\Observer\PredispatchAdminActionControllerObserver

zakdma committed:
MC-24170: Fix Skipped MFTF Tests From MC-17140: MC-13493, MC-14062, MC-14063

ptylek committed:
Add to Compare link does not show in mobile view under 640px in blank theme

arnobsh committed:
MC-30431: MFTF test incorrect behavior after its' implementation in MC-20229

- fixing MFTF test

magento-engcom-team committed:
ENGCOM-6626: magento/magento2#: Unit test for \Magento\AdminNotification\Observer\PredispatchAdminActionControllerObserver #26402

 - Merge Pull Request magento/magento2#26402 from atwixfirster/magento2:unittest-PredispatchAdminActionControllerObserver
 - Merged commits:
   1. 504bfe5f2f10c5003f6155c07bcb29bccf7bedc9

magento-engcom-team committed:
ENGCOM-6627: Issue-25968 - Added additional checking for returning needed variable… #26313

 - Merge Pull Request magento/magento2#26313 from AndreyChorniy/magento2:adjustments-for-correct-variable-type
 - Merged commits:
   1. 0b7a99d7a330216a829056af1b7eb876c14657db
   2. 28e8e5aa46ffe2cfed80124bb95d3ce025a127c8
   3. 71da244811484747f55bf3ccb1244844ff903fd6

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25250

magento-engcom-team committed:
ENGCOM-6628: Add to Compare link does not show in mobile view under 640px in blank theme #26424

 - Merge Pull Request magento/magento2#26424 from ptylek/magento2:show-compare-mobileview-blank
 - Merged commits:
   1. 9ec1d67e18d0b4fed5cc1ef3c8b9a8ad7327d41c

magento-engcom-team committed:
ENGCOM-6593: Add to Compare link not showing in mobile view under 640px #26365

 - Merge Pull Request magento/magento2#26365 from tejash-wagento/magento2:show-compare-mobileview
 - Merged commits:
   1. f4f61f7265c16446ea2444bac6e867192580dce6

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25254

SeruyV committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-29864

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25138

# Conflicts:
#	app/code/Magento/Elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchResultApplier.php

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-24170

zakdma committed:
MC-24170: Fix Skipped MFTF Tests From MC-17140: MC-13493, MC-14062, MC-14063

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25138

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25250

SeruyV committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-29864

viktorpetryk committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-25254

avattam06 committed:
MC-30496: Forward Port [WebAPI Test] Failing Test: Magento.GraphQl.Quote.Customer.GetCustomerCartTest.testGetCustomerCartSecondStore

- Added test changes

avattam06 committed:
MC-30496: Forward Port [WebAPI Test] Failing Test: Magento.GraphQl.Quote.Customer.GetCustomerCartTest.testGetCustomerCartSecondStore

- Fixed static failures

avattam06 committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-30496-test

VladimirZaets committed:
ENGCOM-6628: Add to Compare link does not show in mobile view under 640px in blank theme #26424

VladimirZaets committed:
ENGCOM-6627: Issue-25968 - Added additional checking for returning needed variable… #26313

VladimirZaets committed:
ENGCOM-6626: magento/magento2#: Unit test for \Magento\AdminNotification\Observer\PredispatchAdminActionControllerObserver #26402

VladimirZaets committed:
ENGCOM-6593: Add to Compare link not showing in mobile view under 640px #26365

atwixfirster committed:
magento/magento2#: DeletePaymentToken. Remove redundant validation logic. Test coverage.

avattam06 committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-30496-test

atwixfirster committed:
magento/magento2#: RevokeCustomerToken. Test coverage.

magento-engcom-team committed:
ENGCOM-6636: magento/magento2#: GraphQl. DeletePaymentToken. Remove redundant validation logic. Test coverage. #26452

 - Merge Pull Request magento/magento2#26452 from atwixfirster/magento2:graphql-DeletePaymentToken
 - Merged commits:
   1. 33ea5358c56e0768c62bf37ba44b75a509e67c5a

magento-engcom-team committed:
ENGCOM-6637: magento/magento2#: GraphQl. RevokeCustomerToken. Test coverage. #26464

 - Merge Pull Request magento/magento2#26464 from atwixfirster/magento2:unittest-revokeCustomerToken
 - Merged commits:
   1. 1067926e468c6a630800b1b2f567bffdf58888b7

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

- fixed
- modified tests
- remove redundant ImageProvider

zakdma committed:
Merge remote-tracking branch 'origin/MC-25138' into 2.4-develop-pr5

zakdma committed:
Merge remote-tracking branch 'origin/MC-25250' into 2.4-develop-pr5

zakdma committed:
Merge remote-tracking branch 'origin/MC-25254' into 2.4-develop-pr5

zakdma committed:
Merge remote-tracking branch 'origin/MC-29864' into 2.4-develop-pr5

zakdma committed:
MC-30189: Error Import exported .csv file with small profile generated data

zakdma committed:
Merge remote-tracking branch 'origin/MC-30189' into 2.4-develop-pr5

Commit:
Merge remote-tracking branch 'origin/MC-24170' into 2.4-develop-pr5

Commit:
Fix confusing phpdoc in curl client

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

- static

Commit:
Merge remote-tracking branch 'mainline/2.4-develop' into MC-30255

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into 2.4-develop-pr5

Commit:
Merge remote-tracking branch 'tsg/MC-29745' into Chaika-PR24-2020-01-22

Commit:
Merge remote-tracking branch 'mpi/MC-30431' into Chaika-PR24-2020-01-22

Nazar65 committed:
fix static tests

zakdma committed:
MC-24170: Fix Skipped MFTF Tests From MC-17140: MC-13493, MC-14062, MC-14063

Nazar65 committed:
fix typo

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into 2.4-develop-pr5

zakdma committed:
Merge remote-tracking branch 'origin/MC-24170' into 2.4-develop-pr5

zakdma committed:
MC-24170: Fix Skipped MFTF Tests From MC-17140: MC-13493, MC-14062, MC-14063

zakdma committed:
Merge remote-tracking branch 'origin/MC-24170' into 2.4-develop-pr5

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
ENGCOM-6660: Fix confusing phpdoc in curl client #26495

 - Merge Pull Request magento/magento2#26495 from tdgroot/magento2:fix_confusing_phpdoc_curl
 - Merged commits:
   1. c885c9fbf99c672a06222a33a9b123108ae2ccf9
   2. 613edfc40e02bcaaab12a7fc517a6241c15da72b
   3. 2da2d49878d31f460aad91b542340a4c3dcbd863

lenaorobei committed:
ENGCOM-6660: Fix confusing phpdoc in curl client #26495

lenaorobei committed:
ENGCOM-6637: magento/magento2#: GraphQl. RevokeCustomerToken. Test coverage. #26464

lenaorobei committed:
ENGCOM-6636: magento/magento2#: GraphQl. DeletePaymentToken. Remove redundant validation logic. Test coverage. #26452

lenaorobei committed:
ENGCOM-6565: MFTF: Extract Action Groups to separate files for dev/tests #26322

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

Commit:
Merge remote-tracking branch 'mainline/2.4-develop' into MC-30255

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

- static

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

- static. undefined property

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into 2.4-develop-pr5

Commit:
Merge branch 'MC-29167' of https://github.com/magento-tango/magento2ce into Tango-PR-01-24-2020-24

Commit:
Merge branch 'MC-30260' of https://github.com/magento-tango/magento2ce into Tango-PR-01-24-2020-24

IvanPletnyov committed:
MC-24894: Admin: Check Product categories indexing when add/remove category in product using category link management

lenaorobei committed:
Merge branch '2.4-develop' into 2.4-develop-expedited-prs

Commit:
MC-30255: In Stock Alert Email Shows Incorrect Item Pricing

- added fallback

lenaorobei committed:
ENGCOM-6641: Magento partners issue/PR templates

ajithkumar-maragathavel committed:
Covered validation with MFTF test

ajithkumar-maragathavel commented:
@rogyar As suggested, covered the code changes with MFTF test. Please review it.

ajithkumar-maragathavel committed:
Merge remote-tracking branch 'upstream/2.4-develop' into cms_block_grid_validation

Commit:
Merge remote-tracking branch 'chaika/MC-30255' into Chaika-PR24-2020-01-22

ysapiga committed:
MC-29688: Storefront: Check multiple currencies per websites/store views

dimonovp committed:
MC-30642: Storefront: Configurable product prices

ysapiga committed:
Merge branch 'MC-29688' into 2.4-develop-com-pr5

dimonovp committed:
Merge remote-tracking branch 'origin/MC-30642' into 2.4-develop-com-pr5

avattam06 committed:
Merge remote-tracking branch 'origin/2.4-develop' into MC-30496-test

lenaorobei committed:
Merge branch '2.4-develop' of github.com:magento-engcom/magento2ce into 2.4-develop-expedited-prs

lenaorobei committed:
Merge branch 'issue-pr-templates' of github.com:magento-engcom/magento2ce into 2.4-develop-expedited-prs

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

zakdma committed:
MC-24170: Fix Skipped MFTF Tests From MC-17140: MC-13493, MC-14062, MC-14063

zakdma committed:
Merge remote-tracking branch 'origin/MC-24170' into 2.4-develop-pr5

magento-devops-community-builds-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-pr5

cpartica committed:
Merge pull request #5233 from magento-honey-badgers/MC-30496-test

[honey] MC-30496: Forward Port [WebAPI Test] Failing Test: Magento.GraphQl.Quote.Customer.GetCustomerCartTest.testGetCustomerCartSecondStore

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

magento-mts-svc committed:
merge magento/2.4-develop into magento-chaika/Chaika-PR24-2020-01-22

VladimirZaets committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop latest changes

Accepted Community Pull Requests:
 - magento/magento2#26424: Add to Compare link does not show in mobile view under 640px in blank theme (by @ptylek)
 - magento/magento2#26313: Issue-25968 - Added additional checking for returning needed variable (by @AndreyChorniy)
 - magento/magento2#26402: magento/magento2#: Unit test for \Magento\AdminNotification\Observer\PredispatchAdminActionControllerObserver (by @atwixfirster)
 - magento/magento2#26365: Add to Compare link not showing in mobile view under 640px (by @tejash-wagento)


Fixed GitHub Issues:
 - magento/magento2#26364: Add to Compare link not showing in mobile view under 640px (reported by @tejash-wagento) has been fixed in magento/magento2#26424 by @ptylek in 2.4-develop branch
   Related commits:
     1. 9ec1d67e18d0b4fed5cc1ef3c8b9a8ad7327d41c

 - magento/magento2#25968: `getPrice()` returns a string when setting custom price in admin order (reported by @paul-gene) has been fixed in magento/magento2#26313 by @AndreyChorniy in 2.4-develop branch
   Related commits:
     1. 0b7a99d7a330216a829056af1b7eb876c14657db
     2. 28e8e5aa46ffe2cfed80124bb95d3ce025a127c8
     3. 71da244811484747f55bf3ccb1244844ff903fd6

 - magento/magento2#26364: Add to Compare link not showing in mobile view under 640px (reported by @tejash-wagento) has been fixed in magento/magento2#26365 by @tejash-wagento in 2.4-develop branch
   Related commits:
     1. f4f61f7265c16446ea2444bac6e867192580dce6

Commit:
Merge branch '2.4-develop' of https://github.com/magento/magento2ce into Tango-PR-01-24-2020-24

Commit:
Merge branch 'MC-30431' of https://github.com/magento-mpi/magento2ce into Tango-PR-01-24-2020-24

lenaorobei committed:
Removed CODEOWNERS file

dhorytskyi committed:
Merge pull request #5249 from magento-tango/Tango-PR-01-24-2020-24

Tango PR 01-24-2020 v2.4

magento-mts-svc committed:
merge magento/2.4-develop into magento-engcom/2.4-develop-expedited-prs

magento-mts-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-pr5

zakdma committed:
Merge remote-tracking branch 'origin/MC-21347' into 2.4-develop-pr8

rogyar commented on app/code/Magento/Cms/Test/Mftf/ActionGroup/FillOutBlockContentActionGroup.xml:
```diff
@@ -18,4 +18,14 @@
         <selectOption selector="{{BlockNewPageBasicFieldsSection.storeView}}" userInput="All Store View" stepKey="selectAllStoreView"/>
         <fillField selector="{{BlockContentSection.TextArea}}" userInput="{{_defaultBlock.content}}" stepKey="fillContentField"/>
     </actionGroup>
+    <actionGroup name="FillOutBlockContentWithMarginalSpace">
```
Please, put this action group in a separate file, rename it as `AdminFillCmsBlockFormActionGroup`. Make this action group accept 3 arguments `identifier`, `title` and `content`. And pass the corresponding arguments from your test. 

After this change, you will not need a separate `BlockDataWithMarginalSpace` entity created in `app/code/Magento/Cms/Test/Mftf/Data/BlockPageData.xml`.

Unfortunately, the existing MFTF tests in the core are not the best example of best practices for writing MFTF test. 

Thank you!

ajithkumar-maragathavel replied:
@rogyar Thanks, for the suggestion to improve the code quality. I have made changes to the code as per your suggestions. Please review it.
Please let me know the best practices in MFTF, so that I will also refactor the existing codes in future.

rogyar reviewed (changes requested):
Hi @ajithkumar-maragathavel. Thanks a lot for providing the MFTF test. Please, check my review comment. It needs some minor adjustments to follow the best practices.

Thank you!

ajithkumar-maragathavel committed:
code review changes

naydav committed:
Merge pull request #5251 from magento-engcom/2.4-develop-expedited-prs

[Magento Community Engineering] Community Contributions - 2.4-develop expedited

magento-mts-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-pr5

magento-mts-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-pr8

mykhailomatiola committed:
MC-30722: Layered Navigation with default product attribute "Price"

mykhailomatiola committed:
MC-30722: Layered Navigation with default product attribute "Price"

zakdma committed:
Merge pull request #5271 from magento-tsg/2.4-develop-pr8

[TSG] Fixes for 2.4 (pr8) (2.4-develop)

magento-mts-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-pr5

mykhailomatiola committed:
Merge branch 'MC-30722' into 2.4-develop-com-pr5

Commit:
Merge remote-tracking branch 'mainline/2.4-develop' into Chaika-PR24-2020-01-22

Commit:
Merge branch 'Chaika-PR24-2020-01-22' of https://github.com/magento-chaika/magento2ce into Chaika-PR24-2020-01-22

zakdma committed:
Merge remote-tracking branch 'origin/2.4-develop' into 2.4-develop-com-pr5

zakdma committed:
Merge pull request #5255 from magento-tsg/2.4-develop-pr5

[TSG] Fixes for 2.4 (pr5) (2.4-develop)

magento-mts-svc committed:
merge magento/2.4-develop into magento-chaika/Chaika-PR24-2020-01-22

dhorytskyi committed:
Merge pull request #5253 from magento-chaika/Chaika-PR24-2020-01-22

Chaika-PR24-2020-01-22

magento-mts-svc committed:
merge magento/2.4-develop into magento-tsg/2.4-develop-com-pr5

zakdma committed:
Merge pull request #5276 from magento-tsg/2.4-develop-com-pr5

[TSG-Commerce] Tests for 2.4 (pr5)

ajithkumar-maragathavel committed:
Merge branch 'cms_block_grid_validation' of https://github.com/ajithkumar-maragathavel/magento2-1 into cms_block_grid_validation

rogyar commented on app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminFillCmsBlockFormActionGroup.xml:
```diff
@@ -0,0 +1,25 @@
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="FillOutBlockContentWithMarginalSpace">
```
Hi @ajithkumar-maragathavel. Thank you for your changes. There's one minor adjustment left. The action group name should equal the file name. You named the file correctly but forgot to rename the action group itself.

ajithkumar-maragathavel replied:
@rogyar I have made the changes you suggested. Please review it.
Commit ID - 034f22e120db8332cc7d435a65c20e1279318df7

ajithkumar-maragathavel committed:
Renamed action group

rogyar commented:
Hi @ajithkumar-maragathavel. It looks like you have committed some extra work. Now the PR contains 147 commits. Please, make sure your PR contains only the required changes.

Thank you

ajithkumar-maragathavel commented:
@rogyar Yeah i messed up the branch somehow. I will revert it and commit.

ajithkumar-maragathavel commented:
@rogyar Unfortunately, I deleted my fork accidentally.
Is there a way to restore that? If so please help me.

If not, please let me know what should I do now?
